airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java

```diff
@@ -168,10 +173,21 @@ protected AutoCloseableIterator<JsonNode> queryTableFullRefresh(final BigQueryDa
                                                                   final SyncMode syncMode,
                                                                   final Optional<String> cursorField) {
     LOGGER.info("Queueing query for table: {}", tableName);
-    return queryTable(database, String.format("SELECT %s FROM %s",
-        enquoteIdentifierList(columnNames, getQuoteString()),
-        getFullyQualifiedTableNameWithQuoting(schemaName, tableName, getQuoteString())),
-        tableName, schemaName);
+     // This corresponds to the initial sync for in INCREMENTAL_MODE, where the ordering of the records matters
+    // as intermediate state messages are emitted (if the connector emits intermediate state).
+    if (syncMode.equals(SyncMode.INCREMENTAL) && getStateEmissionFrequency() > 0) {
```

Comment:
why are we doing this here if there is a separate `queryTableIncremental` method?

Reply:
The incremental method is called when there is a state present.
Until that full sync got called (first sync of incremental also run full sync)
So for first full sync we need to arrange elements in order according to cursor field to emit states.
Because select statements not always return data in ascending order of the cursor field.

Reply:
Then lets call this only. why there is an else?